clang/lib/Format/Format.cpp

```diff
@@ -1553,6 +1563,7 @@ FormatStyle getLLVMStyle(FormatStyle::LanguageKind Language) {
   LLVMStyle.AllowShortFunctionsOnASingleLine = FormatStyle::SFS_All;
   LLVMStyle.AllowShortIfStatementsOnASingleLine = FormatStyle::SIS_Never;
   LLVMStyle.AllowShortLambdasOnASingleLine = FormatStyle::SLS_All;
+  LLVMStyle.AllowShortRecordOnASingleLine = FormatStyle::SRS_Never;
```

Comment:
```suggestion
  LLVMStyle.AllowShortRecordOnASingleLine = FormatStyle::SRS_Empty;
```
as `clang-format -style=llvm` gives the following:
```c++
class foo {};

class bar {
  int i;
};
```

Reply:
Making `Empty` the default makes sense but will mess with configs that derive from LLVM, i.e. with this config:
```
BasedOnStyle: LLVM
BreakBeforeBraces: Custom
BraceWrapping:
  AfterClass: true
  SplitEmptyRecord: false
```
We would expect:
```c++
class foo
{};

class bar
{
  int i;
};
```
but get instead:
```
class foo {};

class bar
{
  int i;
};
```
Unsure how to handle this, or if it's a minor enough breaking change that it doesn't matter?

Reply:
If it's set to `Empty` in LLVM style it will be set to `Empty` in this config too, hence the record will be merged without the user knowing about the change. This is why I used `Never` as default and made sure it's not explicitly overriding any existing behavior - making it an opt-in feature.

Reply:
It seems that the current behavior is merging empty records only for `BreakBeforeBraces: Attach`, so I'd add a suboption, say, `EmptyAndAttached`:
- EmptyAndAttached (default)
```c++
class foo {};
struct bar {
  int i;
};
// Or
class foo
{
};
struct bar
{
  int i;
};
```
- Empty (both attached and detached)
```c++
class foo {};
struct bar {
  int i;
};
// Or
class foo {};
struct bar
{
  int i;
}
```